services/ark-cluster-memory/ark-cluster-memory/src/routes/stream.ts

```diff
@@ -22,6 +25,30 @@ const writeSSEEvent = (res: Response, data: StreamResponse): boolean => {
   }
 };
 
+const parseErrorChunk = (error: any): ErrorChunk => {
+  let status = 500
+  let message = ''
+
+  if (typeof error === "string") {
```

Comment:
again, given we know the structure can we use it and then fail if not? rather than trying fallbacks